File: problemset-new/001/00167-two-sum-ii-input-array-is-sorted/freedomDR.go
package main

import "fmt"

func twoSum(numbers []int, target int) []int {
	for i := 0; i < len(numbers)-1; i++ {
		lo, hi, t := i+1, len(numbers), target-numbers[i]
		for lo < hi {
			mid := lo + (hi-lo)/2
			if numbers[mid] < t {
				lo = mid + 1
			} else {
				hi = mid
			}
		}
		if lo != len(numbers) && numbers[lo] == t {
			return []int{i + 1, lo + 1}
		}
	}
	return []int{}
}
func main() {
	fmt.Println("vim-go")
}

File: problemset-new/001/00167-two-sum-ii-input-array-is-sorted/freedomDR.cpp
#include <bits/stdc++.h>

using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int len = numbers.size();
        for(int i = 0; i < len-1; i++) {
            int lo = i+1, hi = len, t = target-numbers[i];
            while( lo < hi ){
                int mid = lo+(hi-lo)/2;
                if(numbers[mid] < t) lo = mid+1;
                else hi = mid;
            }
            if(lo != len && numbers[lo] == t) return {i+1, lo+1};
        }
        return {};
    }
};
int main()
{
    return 0;
}
